test/src/databases/presto/presto.spec.ts

```diff
@@ -0,0 +1,227 @@
+    });
+  });
+
+  it('runs the approx_percentile function', async () => {
+    await expect(`run: presto.sql("""
+      SELECT 1 as n
+      UNION ALL SELECT 50 as n
+      UNION ALL SELECT 100 as n
+      """) -> {
+      aggregate:
+        default_pctl is approx_percentile(n, 0.5)
+        pctl_with_error is approx_percentile(n, .99, 0.1)
+      }`).malloyResultMatches(runtime, {
+      default_pctl: 50,
+      pctl_with_error: 100,
+    });
+  });
+
+  it('runs the bool_and function', async () => {
+    await expect(`run: presto.sql("""
+                SELECT true as n1, false as n2, false as n3
+      UNION ALL SELECT true as n1, true as n2,  false as n3
+      UNION ALL SELECT true as n1, false as n2, false as n3
+      """) -> {
+      aggregate:
+        and_n1 is bool_and(n1)
+        and_n2 is bool_and(n2)
+        and_n3 is bool_and(n3)
+      }`).malloyResultMatches(runtime, {
+      and_n1: true,
+      and_n2: false,
+      and_n3: false,
+    });
+  });
+
+  it('runs the bool_or function', async () => {
+    await expect(`run: presto.sql("""
+                SELECT true as n1, false as n2, false as n3
+      UNION ALL SELECT true as n1, true as n2,  false as n3
+      UNION ALL SELECT true as n1, false as n2, false as n3
+      """) -> {
+      aggregate:
+        or_n1 is bool_or(n1)
+        or_n2 is bool_or(n2)
+        or_n3 is bool_or(n3)
+      }`).malloyResultMatches(runtime, {
+      or_n1: true,
+      or_n2: true,
+      or_n3: false,
+    });
+  });
+
+  it('runs the bitwise_and function', async () => {
+    await expect(`run: presto.sql("""
+                SELECT 13678423 as n1, 23524678 as n2
+      """) -> {
+      select:
+        x is bitwise_and(n1, n2)
+      }`).malloyResultMatches(runtime, {x: 4240710});
+  });
+
+  it('runs the bitwise_or function', async () => {
+    await expect(`run: presto.sql("""
+                SELECT 13678423 as n1, 23524678 as n2
+      """) -> {
+      select:
+        x is bitwise_or(n1, n2)
+      }`).malloyResultMatches(runtime, {x: 32962391});
+  });
+
+  it('runs the variance function', async () => {
+    await expect(`run: presto.sql("""
+      SELECT 1 as n
+      UNION ALL SELECT 50 as n
+      UNION ALL SELECT 100 as n
+      """) -> {
+      aggregate:
+        var is floor(variance(n))
+      }`).malloyResultMatches(runtime, {var: 2450});
+  });
+
+  it('runs the corr function', async () => {
+    await expect(`run: presto.sql("""
+                SELECT 1 as y, 55 as x
+      UNION ALL SELECT 50 as y, 22 as x
+      UNION ALL SELECT 100 as y, 1 as x
+      """) -> {
+      aggregate:
+        correlation is corr(y, x)
+      }`).malloyResultMatches(runtime, {correlation: -0.9911108});
+  });
+
+  it('runs the json_extract_scalar function', async () => {
+    await expect(`run: presto.sql("""
+                SELECT 1 as n
+                """) -> {
+      select:
+        json_arr is json_extract_scalar('[1, 2, 3]', '$[2]')
+        json_obj is json_extract_scalar(
```

Comment:
If you allow the JSON type as mentioned above, you should probably test it here.